### ISSUE TITLE
Integrations: export tool

### ISSUE THREAD

**Issue by PaulUithol** [hotosm/field-campaigner#110]
Ideally, you'd be able to do the following:

- Click on "Export data" button on a specific campaign page
- This would take you to the Export tool, using the correct data model (YAML format) and AOI (as MapCampaign knows both)
- User can select export formats (geopackage, shapefile, etc) and run the export

**Comment by PaulUithol** [comment]
What's the status of this @cchristelis / @smit1678 ? Can we at least link out your current AOI/hopefully data model for easy export?

**Comment by Dragneel7** [comment]
Hi @PaulUithol @smit1678  is this issue still open? I would like to implement this. Can you guide me through the procedure.
Thanks

**Comment by PaulUithol** [comment]
Absolutely, this is something I'd love to have. Roughly, I think we'd need the following components:

- AOI in a format the export tool accepts (geojson?)
- The data model as YAML
- I'd hope the authentication/OAuth can be/is integrated among our tools, so we can use the same OSM token for both tools? If not, this could be a snag and something we'd need to think about across the board (also TM)
- Set up an export job based on this the first time someone clicks "Export", and re-run the existing job on subsequent occurences

This would probably require some coordination with @bdon, @mojodna and @matimahari on the Export tool (https://github.com/hotosm/osm-export-tool/issues). @bdon, do you know if there is an API on the export tool we can use for this?

**Comment by smit1678** [comment]
I think before we fully start working on this we need to outline the flow a bit more. Is this just a function that a campaign manager can only do? Is this just a link to the Export tool or do we use the API to generate everything and a user does not need to go to the Export tool at all? 

@Dragneel7 Can you think propose an idea for what the user flow looks like? 

> @bdon, do you know if there is an API on the export tool we can use for this?

Yes, some of the intro is here: https://export.hotosm.org/en/v3/learn/api.

**Comment by PaulUithol** [comment]
Yup, agreed, we'll def want to sketch this out further. I think a major objective is better integration between our different tools; not necessarily absorbing functionality.

What I think would work well, in terms of workflow, is that we use the API to generate an export job, but then have the "Export data" button link out to the Export tool for that specific job? So the user can adjust parameters, choose export formats as desired (different from defaults we set), etc. Then push the "Run" button & download results.

On API, cool, those docs need some love/work still I guess, perhaps a good opportunity for that as well  ;)

**Comment by PaulUithol** [comment]
Alternatively, that "Export data" button could be multi-part:
- by default, download a shapefile according to default settings. Where we generate the Export tool job, start a run, and link to resulting shapefile in the background. This may be difficult due to async nature of the job though.
- but also offer the opportunity to go over to the specific job page on Export tool to adjust settings?

**Comment by mataharimhairi** [comment]
@smit1678 @PaulUithol Love that you guys are trying to build some connectivity between MapCampaigner and the Export Tool. 

As you mentioned the API and the documentation needs some improving, which perhaps can be revisited with this work!

**Comment by PaulUithol** [comment]
Some comments @Dragneel7:

- I think the "Download JOSM File" button should stay. Editing/reviewing/QA of data (at least that portion of data with warning/errors) is a different workflow from Exporting data
- I would advocate the user should be able to move over to the Export tool, to adjust additional settings to their liking, over trying to integrate all those into MapCampaigner. But the primary option could indeed be a download (shapefile/geopackage) based on default settings and settting up and export using AOI and data model as set up in MapCampaigner.

**Comment by Dragneel7** [comment]
@PaulUithol transporting the GeoJSON file to  export tool would be not be so difficult. Using the export tool sounds cool too. So the only main issue would be to use the access token from OSM OAuth in map campaigner in Export tool to prevent re-login. I am not sure how it is done. Guess this requires some research.

**Comment by timlinux** [comment]
So we will build for now:

* Button in campaigner to take you to export tool
* AOI is set
* List of features required is set

**Comment by timlinux** [comment]
We will do some R&D to see if we can carry the oauth token between the two sites.
